### DISCUSSION TITLE
Prevent GoToAsync from throwing exceptions?

### DISCUSSION THREAD

**Discussion by bzp83** [hardkoded/puppeteer-sharp#2706] [Q&A] [answered]
Is this possible to prevent GoToAsync from throwing exceptions? I need to have access to the error, if any, net::ERR_ABORTED, net::ERR_NAME_NOT_RESOLVED etc.

I'm currently catching the exceptions and using ex.Message.Contains but would prefer to have a proper way of doing this.

thanks

**Comment by kblok** [comment, marked as answer]
You could create an extension method. something like `public static Task<IResponse> TryGoToAsync(this IPage page)`

**Reply by bzp83** [reply]
> You could create an extension method. something like `public static Task<IResponse> TryGoToAsync(this IPage page)`

Thanks for your answer.

I was actually looking for some better way to get the actual error code instead of parsing/checking the error message. I checked the source code and it seems like it's not possible, so it's all good. I'll keep doing what I'm doing.

thanks!
